src/model.ts

```diff
@@ -281,7 +281,7 @@ export class Model extends EventBus<any> implements CommandDispatcher {
 
     this.joinSession();
 
-    if (config.snapshotRequested) {
+    if (config.snapshotRequested || data["[Content_Types].xml"]) {
```

Comment:
it should not snapshot if the spreadsheet is in read-only mode (because the user most cannot update the spreadsheet in any way)